### ISSUE TITLE
Ctrl-C/Ctrl-D in the CPAN shell  causes an endless loop in blead-perl/maint-perl

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#6430]
Migrated from [rt.perl.org#21897](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=21897) (status was 'resolved')

Searchable as RT21897$

**Comment by p5pRT** [comment]
### From stas@stason.org

#### Created by stas@hope.stason.org



The following components were used&#8203;:

CPAN 0\.70
Term&#8203;::ReadLine&#8203;::Perl 1\.0203

In the following tests start the CPAN shell and then press Ctrl\-C\, followed by 
Ctrl\-D

== 5\.8\.0&#8203;:

% perl\-5\.8\.0 \-MCPAN \-eshell
cpan>
Caught SIGINT
cpan> Use of uninitialized value in ord at
/home/stas/perl/5\.8\.0/lib/site\_perl/5\.8\.0/Term/ReadLine/readline\.pm line 1481\.
cpan>

prints one warning and returns the prompt

== blead perl \(and 5\.8\.1\-tobe\)&#8203;:

% perl\-blead \-MCPAN \-eshell
cpan>
Caught SIGINT
cpan> Use of uninitialized value in ord at
/home/stas/perl/5\.8\.0/lib/site\_perl/5\.8\.0/Term/ReadLine/readline\.pm line 1481\.
\[\.\.\.\]
results in endless loop consisting of the last line

same problem for 5\.8\.1\-tobe

Certainly\, the warning is probably a problem in CPAN\.pm\, but the endless loop 
appearing only the blead/maint versions doesn't sound good\.

<details><summary>Perl Info</summary>

```

Flags:
     category=core
     severity=low

Site configuration information for perl v5.9.0:

Configured by stas at Mon Apr  7 10:30:23 EST 2003.

Summary of my perl5 (revision 5.0 version 9 subversion 0 patch 19140) 
configuration:
   Platform:
     osname=linux, osvers=2.4.19-16mdksmp, archname=i686-linux
     uname='linux hope.stason.org 2.4.19-16mdksmp #1 smp fri sep 20 16:08:37 
cest 2002 i686 unknown unknown gnulinux '
     config_args='-des -Dprefix=/home/stas/perl/blead -Doptimize=-g 
-Duseshrplib -Dusedevel -DDEBUG_LEAKING_SCALARS'
     hint=recommended, useposix=true, d_sigaction=define
     usethreads=undef useithreads=undef usemultiplicity=undef
     useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
     use64bitint=undef use64bitall=undef uselongdouble=undef
     usemymalloc=n, bincompat5005=undef
   Compiler:
     cc='cc', ccflags ='-DDEBUGGING -fno-strict-aliasing -D_LARGEFILE_SOURCE 
-D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm',
     optimize='-g',
     cppflags='-DDEBUGGING -fno-strict-aliasing -I/usr/include/gdbm'
     ccversion='', gccversion='3.2 (Mandrake Linux 9.0 3.2-1mdk)', gccosandvers=''
     intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
     d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
     ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', 
lseeksize=8
     alignbytes=4, prototype=define
   Linker and Libraries:
     ld='cc', ldflags =' -L/usr/local/lib'
     libpth=/usr/local/lib /lib /usr/lib
     libs=-lnsl -lgdbm -ldb -ldl -lm -lc -lcrypt -lutil -lrt
     perllibs=-lnsl -ldl -lm -lc -lcrypt -lutil -lrt
     libc=/lib/libc-2.3.1.so, so=so, useshrplib=true, libperl=libperl.so
     gnulibc_version='2.3.1'
   Dynamic Linking:
     dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic 
-Wl,-rpath,/home/stas/perl/blead/lib/5.9.0/i686-linux/CORE'
     cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
     DEVEL18374


@INC for perl v5.9.0:
     /home/stas/perl/blead/lib/5.9.0/i686-linux
     /home/stas/perl/blead/lib/5.9.0
     /home/stas/perl/blead/lib/site_perl/5.9.0/i686-linux
     /home/stas/perl/blead/lib/site_perl/5.9.0
     /home/stas/perl/blead/lib/site_perl
     .


Environment for perl v5.9.0:
     HOME=/home/stas
     LANG=en_US
     LANGUAGE=en_US:en
     LC_COLLATE=en_US
     LC_CTYPE=en_US
     LC_MESSAGES=en_US
     LC_MONETARY=en_US
     LC_NUMERIC=en_US
     LC_TIME=en_US
     LD_LIBRARY_PATH=/usr/X11R6/lib:/usr/X11R6/lib/modules
     LOGDIR (unset)
 
PATH=/usr//bin:/bin:/usr/bin:.:/usr/local/bin:/usr/X11R6/bin:/usr/games:/home/stas/bin:/home/stas/bin:/usr/local/bin:/usr/X11R6/bin:/usr/java/j2re1.4.0/bin/
     PERL_BADLANG (unset)
     SHELL=/bin/tcsh

__________________________________________________________________
Stas Bekman            JAm_pH ------> Just Another mod_perl Hacker
http://stason.org/     mod_perl Guide ---> http://perl.apache.org
mailto:stas@stason.org http://use.perl.org http://apacheweek.com
http://modperlbook.org http://apache.org   http://ticketmaster.com


```
</details>

**Comment by p5pRT** [comment]
### From @smpeters

> \[stas \- Tue Apr 08 22&#8203;:09&#8203;:49 2003\]&#8203;:
> 
> 
> This is a bug report for perl from stas@&#8203;hope\.stason\.org\,
> generated with the help of perlbug 1\.34 running under perl v5\.9\.0\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> The following components were used&#8203;:
> 
> CPAN 0\.70
> Term&#8203;::ReadLine&#8203;::Perl 1\.0203
> 
> In the following tests start the CPAN shell and then press Ctrl\-C\,
> followed by
> Ctrl\-D
> 
> == 5\.8\.0&#8203;:
> 
> % perl\-5\.8\.0 \-MCPAN \-eshell
> cpan>
> Caught SIGINT
> cpan> Use of uninitialized value in ord at
> /home/stas/perl/5\.8\.0/lib/site\_perl/5\.8\.0/Term/ReadLine/readline\.pm
> line 1481\.
> cpan>
> 
> prints one warning and returns the prompt
> 
> == blead perl \(and 5\.8\.1\-tobe\)&#8203;:
> 
> % perl\-blead \-MCPAN \-eshell
> cpan>
> Caught SIGINT
> cpan> Use of uninitialized value in ord at
> /home/stas/perl/5\.8\.0/lib/site\_perl/5\.8\.0/Term/ReadLine/readline\.pm
> line 1481\.
> \[\.\.\.\]
> results in endless loop consisting of the last line
> 
> same problem for 5\.8\.1\-tobe
> 
> Certainly\, the warning is probably a problem in CPAN\.pm\, but the
> endless loop
> appearing only the blead/maint versions doesn't sound good\.
> 

I was able to replicate this in the current blead/5\.8\.6\-tobe after
installing Term&#8203;::ReadLine&#8203;::Perl\.  However\, after replacing
Term&#8203;::ReadLine&#8203;::Perl with Term&#8203;::ReadLine&#8203;::Gnu\, it worked just fine\.  I'm
thinking Term&#8203;::ReadLine&#8203;::Perl is more the culprit than CPAN\.

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From stas@stason.org

Steve Peters via RT wrote&#8203;:
\[\.\.\.\]
> I was able to replicate this in the current blead/5\.8\.6\-tobe after
> installing Term&#8203;::ReadLine&#8203;::Perl\.  However\, after replacing
> Term&#8203;::ReadLine&#8203;::Perl with Term&#8203;::ReadLine&#8203;::Gnu\, it worked just fine\.  I'm
> thinking Term&#8203;::ReadLine&#8203;::Perl is more the culprit than CPAN\.

So may be redirect that report to Term&#8203;::ReadLine&#8203;::Perl module then?

\-\- 
\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_
Stas Bekman            JAm\_pH \-\-\-\-\-\-> Just Another mod\_perl Hacker
http&#8203;://stason\.org/     mod\_perl Guide \-\-\-> http&#8203;://perl\.apache\.org
mailto&#8203;:stas@&#8203;stason\.org http&#8203;://use\.perl\.org http&#8203;://apacheweek\.com
http&#8203;://modperlbook\.org http&#8203;://apache\.org   http&#8203;://ticketmaster\.com

**Comment by p5pRT** [comment]
### From @smpeters

> \[stas \- Thu Nov 04 14&#8203;:28&#8203;:30 2004\]&#8203;:
> 
> Steve Peters via RT wrote&#8203;:
> \[\.\.\.\]
> > I was able to replicate this in the current blead/5\.8\.6\-tobe after
> > installing Term&#8203;::ReadLine&#8203;::Perl\.  However\, after replacing
> > Term&#8203;::ReadLine&#8203;::Perl with Term&#8203;::ReadLine&#8203;::Gnu\, it worked just fine\.  I'm
> > thinking Term&#8203;::ReadLine&#8203;::Perl is more the culprit than CPAN\.
> 
> So may be redirect that report to Term&#8203;::ReadLine&#8203;::Perl module then?
> 
> 

A ticket on this problem with Term&#8203;::ReadLine&#8203;::Perl has been opened at
rt\.cpan\.org and is available at http&#8203;://rt\.cpan\.org/NoAuth/Bug\.html?id=8587

**Comment by p5pRT** [comment]
@smpeters - Status changed from 'open' to 'resolved'
